src/Identity/Documents/InternationalPassport.php

```diff
@@ -8,251 +10,169 @@
+     * @var int|null The year (e.g. 1997) that the passport was issued.
      */
     private ?int $IssueYear = null;
 
     /**
-     * @var string|null The full country name (e.g. "New Zealand", not "NZ") that this passport was issued by
+     * @var string|null The full country name (e.g. "New Zealand", not "NZ") that this passport was issued by.
      */
     private ?string $CountryOfOrigin = null;
 
-    /**
-     * @return string|null
-     */
     public function getNumber(): ?string
     {
         return $this->Number;
     }
 
-    /**
-     * @param string|null $Number
-     *
-     * @return InternationalPassport
-     */
     public function setNumber(?string $Number): InternationalPassport
     {
         $this->Number = $Number;
 
         return $this;
     }
 
-    /**
-     * @return string|null
-     */
     public function getShortPassportNumber(): ?string
     {
         return $this->ShortPassportNumber;
     }
 
-    /**
-     * @param string|null $ShortPassportNumber
-     *
-     * @return InternationalPassport
-     */
     public function setShortPassportNumber(?string $ShortPassportNumber): InternationalPassport
     {
         $this->ShortPassportNumber = $ShortPassportNumber;
 
         return $this;
     }
 
-    /**
-     * @return int|null
-     */
     public function getExpiryDay(): ?int
     {
         return $this->ExpiryDay;
     }
 
-    /**
-     * @param int|null $ExpiryDay
-     *
-     * @return InternationalPassport
-     */
     public function setExpiryDay(?int $ExpiryDay): InternationalPassport
     {
         $this->ExpiryDay = $ExpiryDay;
 
         return $this;
     }
 
-    /**
-     * @return int|null
-     */
     public function getExpiryMonth(): ?int
     {
         return $this->ExpiryMonth;
     }
 
-    /**
-     * @param int|null $ExpiryMonth
-     *
-     * @return InternationalPassport
-     */
     public function setExpiryMonth(?int $ExpiryMonth): InternationalPassport
     {
         $this->ExpiryMonth = $ExpiryMonth;
 
         return $this;
     }
 
-    /**
-     * @return int|null
-     */
     public function getExpiryYear(): ?int
     {
         return $this->ExpiryYear;
     }
 
-    /**
-     * @param int|null $ExpiryYear
-     *
-     * @return InternationalPassport
-     */
     public function setExpiryYear(?int $ExpiryYear): InternationalPassport
     {
         $this->ExpiryYear = $ExpiryYear;
 
         return $this;
     }
 
-    /**
-     * @param DateTime|null $expiryDate
-     *
-     * @return InternationalPassport
-     */
     public function setExpiryDate(?DateTime $expiryDate): InternationalPassport
     {
-        $this->setExpiryDay($expiryDate->format('d'));
-        $this->setExpiryMonth($expiryDate->format('m'));
-        $this->setExpiryYear($expiryDate->format('Y'));
+        $this->setExpiryDay((int)$expiryDate->format('d'));
+        $this->setExpiryMonth((int)$expiryDate->format('m'));
+        $this->setExpiryYear((int)$expiryDate->format('Y'));
```

Comment:
A null check on the `$expiryDate` is necessary before these lines.

Reply:
Good point. Looking back at the history of the method this previously took separate values that were all required ints, I think rather than adding null checks (and then either returning `null` or throwing an `Exception`, neither of which feels like a good option for a setter method), I think we should just enforce that DateTime exists at a language level by making it non-nullable.

I've made that change instead, let me know if you have a use case where you want to pass in a null `DateTime`?

FYI this is technically a breaking change but I think given how quickly we are releasing these new versions and the likelihood that this is actually being used in such a way it's unlikely, I won't release a v2.0.0 just for this fix.

Reply:
I made DateTime fields to be nullable for the sake of consistency, so you can set the fields with a same logic confidently. I made the decision also based on a fact that date fields are nullable as per the WSDL.

Reply:
Alright, that makes sense to me - I've configured it so that if the value is null then it resets all fields that actually get sent to ID3global to null, I think this makes the most sense (rather than e.g. returning after doing nothing or throwing an Exception) - what do you think?

Reply:
Yes, you are absolutely right. I did not take into account the reset logic.

Reply:
Cool, I'll resolve this conversation now then. Will look to merge this once I can test it on at least the pilot environment (my account is currently disabled :))